packages/api/models/granules.js

```diff
@@ -64,29 +64,48 @@ class Granule extends Manager {
    * @returns {Promise} an object showing the start of the re-ingest
    */
   async reingest(g) {
+    await this.applyWorkflow(g, 'IngestGranule', 'input');
+    return {
+      granuleId: g.granuleId,
+      action: 'reingest',
+      status: 'SUCCESS'
+    };
+  }
+
+  /**
+   * apply a workflow to a given granule object
+   *
+   * @param {Object} g - the granule object
+   * @param {string} workflow - the workflow name
+   * @param {string} messageSource - 'input' or 'output' from previous execution
+   * @param {Object} metaOverride - overrides the meta of the new execution
+   * @param {Object} payloadOverride - overrides the payload of the new execution
+   * @returns {Promise} an object showing the start of the workflow execution
+   */
+  async applyWorkflow(g, workflow, messageSource, metaOverride, payloadOverride) {
     const { name, version } = deconstructCollectionId(g.collectionId);
 
     // get the payload of the original execution
     const status = await aws.StepFunction.getExecutionStatus(path.basename(g.execution));
-    const originalMessage = JSON.parse(status.execution.input);
+    const originalMessage = JSON.parse(status.execution[messageSource]);
```

Comment:
We probably should return a meaningful error if the execution is not found (like after 90 days) or for any other reasons.

Reply:
I'm wondering if we really need the previous execution.  The granule itself has all the files and collection information, and can we build the payload from the granule?

Reply:
I think we will need the previous execution for reingest, since the payload for the remote files comes from ParsePDR (for the PDR workflow, at least). For in-place it might be different.